### PR TITLE
Sig 2299 custom select component

### PR DESCRIPTION
**Pull request by jpoppe**
@janjaap please pre-review, thanks!

### CODE REVIEW

File: src/signals/settings/users/containers/Overview/index.js [hunk 1]
```diff
@@ -128,6 +140,23 @@ export const UsersOverviewContainer = ({ userCan }) => {
 
   const columnHeaders = ['Gebruikersnaam', 'Rol', 'Status'];
 
+  const userActiveFilter = createOnChangeFilter('is_active');
+  const roleFilter = createOnChangeFilter('role');
+
+  const userStatusOnChangeHandler = event => {
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/index.js]: `useCallback`

File: src/signals/settings/users/containers/Overview/index.js [hunk 2]
```diff
@@ -128,6 +140,23 @@ export const UsersOverviewContainer = ({ userCan }) => {
...
+    event.preventDefault();
+
+    setUserActiveState(event.target.value);
+    userActiveFilter(event.target.value);
+  };
+
+  const roleOnChangeHandler = event => {
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/index.js]: `useCallback`

**Comment by jpoppe** [comment]
@janjaap I have updated the tests with fetch fixtures, could you please check if this is what you meant? It still feels like a bit like testing in the dark for me. Thanks!

File: src/signals/settings/users/containers/Overview/index.js [hunk 3]
```diff
@@ -88,22 +97,26 @@ const UsersOverviewContainer = () => {
...
-        currentTarget: {
-          dataset: { itemId },
-        },
-      } = e;
+      const { currentTarget: { dataset: { itemId } } } = event;
 
-      if (itemId) {
-        history.push(`${USER_URL}/${itemId}`);
-      }
+      if (itemId) { history.push(`${USER_URL}/${itemId}`); }
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/index.js]: Does the linter turn this into a one-liner?

File: src/signals/settings/users/containers/Overview/hooks/useFetchUsers.js [hunk 4]
```diff
@@ -36,18 +36,18 @@ const useFetchUsers = ({ page, filters } = {}) => {
 
       const pageParams = [page && `page=${page}`, `page_size=${PAGE_SIZE}`];
 
+
       const filterParams = Object.entries(filters || {})
-        .filter(([, value]) => Boolean(value))
+        .filter(([, value]) => !['*'].includes(value))
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/hooks/useFetchUsers.js]: Why not
```jsx
.filter(([, value]) => value !== '*')
```
Bit easier to read, wouldn't you agree?

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 5]
```diff
@@ -440,4 +422,176 @@ describe('signals/settings/users/containers/Overview', () => {
...
+    // check if the default value has been set
+    expect(filterByRoleSelect.value).toBe('*');
+
+    const filterValue = 'Regievoerder';
+
+    act(() => {
+      fireEvent.change(filterByRoleSelect, { target: { value: filterValue } });
+    });
+
+    expect(dispatch).toHaveBeenCalledTimes(1);
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Can you assert that `dispatch` has not been called before the event was fired? For good measure.

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 6]
```diff
@@ -440,4 +422,176 @@ describe('signals/settings/users/containers/Overview', () => {
...
+    // check if the default value has been set
+    expect(filterByUserActiveSelect.value).toBe('*');
+
+    const filterValue = 'true';
+
+    act(() => {
+      fireEvent.change(filterByUserActiveSelect, { target: { value: filterValue } });
+    });
+
+    expect(dispatch).toHaveBeenCalledTimes(1);
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Same comment here; can you assert that `dispatch` hasn't been called prior to firing the event?

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 7]
```diff
@@ -440,4 +422,176 @@ describe('signals/settings/users/containers/Overview', () => {
...
+    expect(dispatch).toHaveBeenCalledTimes(2);
+
+    expect(dispatch).toHaveBeenCalledWith(setUserFilters({ is_active: userActiveFilterValue }));
+    expect(dispatch).toHaveBeenCalledWith(setUserFilters({ role: roleFilterValue }));
+
+    expect(filterByUserActiveSelect.value).toBe(userActiveFilterValue);
+    expect(filterByRoleSelect.value).toBe(roleFilterValue);
+  });
+
+  it('dispatches the correct actions on successful getSeason fetch request', async () => {
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: getSeason?

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 8]
```diff
@@ -440,4 +422,176 @@ describe('signals/settings/users/containers/Overview', () => {
...
+
+    await wait();
+
+    fetch.mockResponses(
+      [JSON.stringify(usersFiltersDisabledJSON), { status: 200 }],
+      [JSON.stringify(usersFiltersActiveJSON), { status: 200 }],
+      [JSON.stringify(usersFilterInactiveJSON), { status: 200 }],
+      [JSON.stringify(usersFiltersActiveAndRoleJSON), { status: 200 }],
+      [JSON.stringify(usersFiltersInactiveAndRoleJSON), { status: 200 }],
+    );
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Naaisch

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 9]
```diff
@@ -440,4 +422,176 @@ describe('signals/settings/users/containers/Overview', () => {
...
+
+    const filterByUserActiveSelect = getByTestId('userActiveSelect');
+    const filterByRoleSelect = getByTestId('roleSelect');
+    const apiPrefix = 'https://acc.api.data.amsterdam.nl/signals/v1/private/users?page=1&page_size';
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Maybe get this from the `configuration` class?

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 10]
```diff
@@ -440,4 +422,176 @@ describe('signals/settings/users/containers/Overview', () => {
...
+
+    // filters: disabled
+    expect(filterByUserActiveSelect.value).toBe('*');
+    expect(filterByRoleSelect.value).toBe('*');
+
+    const filtersDisabled = await fetch(apiPrefix);
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Why would you need to `fetch` here if you can just use the JSON form the fixture?

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 11]
```diff
@@ -11,15 +11,24 @@ import {
...
-import UsersOverview from '..';
+import { setUserFilters } from 'signals/settings/actions';
+import SettingsContext from 'signals/settings/context';
+import * as rolesSelectors from 'models/roles/selectors';
+
+import usersFilterInactiveJSON from 'utils/__tests__/fixtures/usersFilterInactive.json';
+import usersFiltersActiveJSON from 'utils/__tests__/fixtures/usersFiltersActive.json';
+import usersFiltersActiveAndRoleJSON from 'utils/__tests__/fixtures/usersFiltersActiveAndRole.json';
+import usersFiltersDisabledJSON from 'utils/__tests__/fixtures/usersFiltersDisabled.json';
+import usersFiltersInactiveAndRoleJSON from
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Can't you use one fixture and perform filters on that set in this test? Then we don't need as many fixtures.
🤷‍♂

Comment (jpoppe) [reply src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Will do.

File: src/signals/settings/users/containers/Overview/index.js (hunk 3 above)
Comment (jpoppe) [reply src/signals/settings/users/containers/Overview/index.js]: My linter in my mind does, such a short line which took 3 lines, I would even prefer not to use the curly braces and semicolon.

Will change it back..

File: src/signals/settings/users/containers/Overview/hooks/useFetchUsers.js (hunk 4 above)
Comment (jpoppe) [reply src/signals/settings/users/containers/Overview/hooks/useFetchUsers.js]: The reason I used the includes is that I filtered for multiple values in the first place, indeed not needed.

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js (hunk 7 above)
Comment (jpoppe) [reply src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Oops..

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js (hunk 8 above)
Comment (jpoppe) [reply src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Thanks!

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 12]
```diff
@@ -82,15 +81,9 @@ describe('signals/settings/users/containers/Overview', () => {
...
-        Accept: 'application/json',
-      },
-    };
-    const history = {
-      ...memoryHistory,
-      push,
-    };
+    const apiHeaders = { headers: { Accept: 'application/json' } };
+
+    const history = { ...memoryHistory, push };
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Might be good to set the dispatch method between tests

```jsx
dispatch.mockReset();
```

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 13]
```diff
@@ -440,4 +417,178 @@ describe('signals/settings/users/containers/Overview', () => {
...
+    expect(filtersDisabledResult.count).toBe(103);
+
+    // filters: user active = true
+    const userActiveFilterValue = 'true';
+    act(() => {
+      fireEvent.change(filterByUserActiveSelect, { target: { value: userActiveFilterValue } });
+    });
+    expect(dispatch).toHaveBeenCalledTimes(1);
+
+    const filtersActive = await fetch(`${apiPrefix}&is_active=true`);
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]:
```jsx
expect(fetch).toHaveBeenCalledWith(expect.stringContaining('?page=1&skdfksfsdf')));
```

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js (hunk 9 above)
Comment (jpoppe) [reply src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Not needed anymore in the new implementation.

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js (hunk 10 above)
Comment (jpoppe) [reply src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Also not needed anymore in the new implementation.

File: src/signals/settings/users/containers/Overview/__tests__/Overview.test.js [hunk 14]
```diff
@@ -521,74 +521,104 @@ describe('signals/settings/users/containers/Overview', () => {
...
+
+    expect(fetch).toHaveBeenCalledWith(
+      expect.stringContaining('role=Behandelaar'),
+      expect.objectContaining(testContext.apiHeaders)
+    );
+
+    expect(fetch).toHaveBeenCalledWith(
+      expect.stringContaining('is_active=true'),
+      expect.objectContaining(testContext.apiHeaders)
+    );
```
Comment (janjaap) [inline src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: I would combine these two into one assertion. Now it looks like there have been two fetch calls.
Also, assets the amount of fetch calls. Should've only been 1

Comment (jpoppe) [reply src/signals/settings/users/containers/Overview/__tests__/Overview.test.js]: Done :)

**Pull request merged**